crypto/provider_core.c

```diff
@@ -527,6 +527,12 @@ static int provider_activate(OSSL_PROVIDER *prov)
     return 1;
 }
 
+int ossl_provider_available(OSSL_PROVIDER *prov)
+{
+    return prov != NULL
+        && (prov->flag_initialized || prov->flag_fallback);
```

Comment:
So what happens if other providers are loaded but `prov` is a fallback provider that hasn't been initialized? Doesn't that mean `ossl_provider_available` will return true even though the provider isn't currently loaded and won't ever be with an explicit call to `OSSL_PROVIDER_load`?